### PR TITLE
Make push-build.sh independent

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Before removing anago, we have to ensure that the legacy script
`push-build.sh` is independent from the `lib/` directory.

What we do with this patch is plainly copying the previously sourced
dependencies into the `push-build.sh` script while removing unused
functions. Unused variables have not been cleaned-up.
#### Which issue(s) this PR fixes:

Refers to #1673 
#### Special notes for your reviewer:
This is hard to test and the last time we broke a test. @hasheddan, do you know any way to pre-test this change?
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

@saschagrunert you are right that there isn't a great way to test this. In the past I have mostly just run it locally with certain actions commented out or early exist. This obviously is not foolproof because we are literally modifying the code to test it. However, this is a fairly mechanical change here so I think we can move forward and just keep an eye on testgrid for the next few days.

/hold

I'll let you decide when you feel good about it :)

**Comment by saschagrunert** [comment]
Okay cool, let's cancel the hold. :see_no_evil: 
/unhold

**Pull request merged**
